lib/lib/utility/calendar.js

```diff
@@ -89,16 +85,82 @@ function calculateTimeForXPosition(canvasTimeStart, canvasTimeEnd, canvasWidth,
   return timeFromCanvasTimeStart + canvasTimeStart;
 }
 
+function setUnitValue(value, unit, callback) {
+  var date = new Date(value);
+
+  switch (unit) {
+    case 'year':
+      date.setYear(callback(date.getFullYear()));
+      break;
+
+    case 'month':
+      date.setMonth(callback(date.getMonth()));
+      break;
+
+    case 'week': // ???
+
+    case 'day':
+      date.setDate(callback(date.getDate()));
+      break;
+
+    case 'hour':
+      date.setHours(callback(date.getHours()));
+      break;
+
+    case 'minute':
+      date.setMinutes(callback(date.getMinutes()));
+      break;
+
+    case 'seconds':
+      date.setSeconds(callback(date.getSeconds()));
+      break;
+  }
+
+  return date;
+}
+
+function setUnitStart(date, unit) {
+  date = new Date(date);
+
+  switch (unit) {
+    case 'year':
+      date.setMonth(0);
+
+    case 'month':
+      date.setDate(1);
+
+    case 'week': // ???
+
+    case 'day':
+      date.setHours(0);
+
+    case 'hour':
+      date.setMinutes(0);
+
+    case 'minute':
+      date.setSeconds(0);
+
+    case 'seconds':
+      date.setMilliseconds(0);
+  }
+
+  return date;
+} // | `unit`| `second`, `minute`, `hour`, `day`, `week`, `month`, `year` intervals between columns |
```

Comment:
It is the third time we use this structure, it could be interesting to turn it into a util function I think.

Reply:
[improve utility fcts](https://github.com/Optiways/react-calendar-timeline/pull/1/commits/a30c37ad347b34dfa524e63e2c61d389184123de)